### ISSUE TITLE
Request invalid for Wear platform

### ISSUE THREAD

**Issue by vijai1996** [ivanmeler/SamFirm_Reborn#3]
The new Galaxy Watch 4 is based on wear OS and Samsung uses the same method as Galaxy phones to check for updates and download. Samfirm succeeds in checking the latest firmware for models but download fails with result code 400. I'm not sure what has changed in the download request but if you could guide me what you need, I might be able to help you with information.

Sample model:
Model number: SM-R885F
CSC: INU

**Comment by ivanmeler** [comment]
I noticed the same thing when trying to download fw for it, as of right now it seems they dont use same method to fetch firmwares (listing them is the same), Ill need to investigate this further, Best bet would be trying to download fw with official tools from samsung to see what requests they send and where

**Comment by vijai1996** [comment]
I had the same thought but for now, only way to check and download update is through the companion plugin on phone which means, packet sniffing is not possible without root as far as I know.

**Comment by ivanmeler** [comment]
> I had the same thought but for now, only way to check and download update is through the companion plugin on phone which means, packet sniffing is not possible without root as far as I know.

Yeah thing with those is that they are not full updates like what samfirm fetches but just update.zip files that get applied on top of base rom

**Comment by vijai1996** [comment]
> > I had the same thought but for now, only way to check and download update is through the companion plugin on phone which means, packet sniffing is not possible without root as far as I know.
> 
> Yeah thing with those is that they are not full updates like what samfirm fetches but just update.zip files that get applied on top of base rom

Exactly, those are delta updates. I'm not sure but I thought fetching delta and full package uses same end point but different request param? Am I wrong?

**Comment by ivanmeler** [comment]
> > > I had the same thought but for now, only way to check and download update is through the companion plugin on phone which means, packet sniffing is not possible without root as far as I know.
> > 
> > 
> > Yeah thing with those is that they are not full updates like what samfirm fetches but just update.zip files that get applied on top of base rom
> 
> Exactly, those are delta updates. I'm not sure but I thought fetching delta and full package uses same end point but different request param? Am I wrong?

unfortunately they dont use same servers for full updates and deltas, only good thing about delta updates is they dont require any special auth to download like full packages do

**Comment by mirh** [comment]
https://github.com/zacharee/SamloaderKotlin/pull/422
https://github.com/topjohnwu/samloader-rs/pull/1
